codex-rs/tui/src/app.rs

```diff
@@ -357,3 +429,119 @@ impl<'a> App<'a> {
+    use codex_core::protocol::AgentMessageEvent;
+    use codex_core::protocol::EventMsg;
+    use codex_core::protocol::TaskCompleteEvent;
+    use pretty_assertions::assert_eq;
+
+    #[test]
+    fn test_create_compact_summary_prompt_with_content() {
+        let summary_text = "User asked about Rust. I explained ownership and borrowing.";
+        let result = create_compact_summary_prompt(summary_text);
+
+        let expected = COMPACT_SUMMARY_TEMPLATE.replace(
+            "{}",
+            "User asked about Rust. I explained ownership and borrowing.",
+        );
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn test_create_compact_summary_prompt_empty_content() {
+        let result = create_compact_summary_prompt("");
+        assert_eq!(result, "Previous conversation has been summarized.");
+
+        let result_whitespace = create_compact_summary_prompt("   \n\t   ");
+        assert_eq!(
+            result_whitespace,
+            "Previous conversation has been summarized."
+        );
+    }
+
+    #[test]
+    fn test_pending_summarization_state_management() {
+        let mut pending = PendingSummarization {
+            summary_buffer: String::new(),
+        };
+
+        // Simulate collecting summary pieces
+        pending.summary_buffer.push_str("First part of summary");
+        pending.summary_buffer.push('\n');
+        pending.summary_buffer.push_str("Second part of summary");
+
+        let expected = "First part of summary\nSecond part of summary";
+        assert_eq!(pending.summary_buffer, expected);
+
+        // Test that create_compact_summary_prompt works with collected buffer
+        let prompt = create_compact_summary_prompt(&pending.summary_buffer);
+        assert!(prompt.contains("First part of summary"));
+        assert!(prompt.contains("Second part of summary"));
+    }
+
+    #[test]
+    fn test_compact_summary_template_integrity() {
+        // Ensure the template has expected structure and placeholder
+        assert!(COMPACT_SUMMARY_TEMPLATE.contains("{}"));
+        assert!(COMPACT_SUMMARY_TEMPLATE.contains("continuation of a previous conversation"));
+        assert!(COMPACT_SUMMARY_TEMPLATE.contains("/compact command has been applied"));
+    }
+
+    #[test]
+    fn test_agent_message_event_creation() {
+        // Test that we can create the events we expect to handle
+        let msg_event = EventMsg::AgentMessage(AgentMessageEvent {
+            message: "This is a test summary".to_string(),
+        });
+
+        if let EventMsg::AgentMessage(agent_msg) = msg_event {
+            assert_eq!(agent_msg.message, "This is a test summary");
+        } else {
+            panic!("Expected AgentMessage event");
+        }
+
+        let task_complete_event = EventMsg::TaskComplete(TaskCompleteEvent {
+            last_agent_message: Some("Final message".to_string()),
+        });
+
+        matches!(task_complete_event, EventMsg::TaskComplete(_));
+    }
+
+    #[test]
```

Comment:
What is this test telling us?

A test should help reduce risk. If the test is implemented as `format!()` as I suggested, then the risk of it failing in any of the ways this verifies seems low.